### DISCUSSION TITLE
When does pruning normally happen?

### DISCUSSION THREAD

**Discussion by Niksko** [argoproj/argo-cd#14364] [Q&A]
In a standard ArgoCD application with no sync waves configured, does pruning happen after new resources have been created?

The docs talk about a 'prune last' setting https://argo-cd.readthedocs.io/en/stable/user-guide/sync-options/#prune-last, but is this feature meant for dealing with custom sync waves? If there is only effectively one sync wave (because sync waves aren't in use at all), can we safely assume that new resources will be created first, and then old resources will be pruned?

Basically, does Argo follow the 'create before delete' pattern?

**Comment by bowale01** [comment]
Based on my experience with ArgoCD, when no sync waves are configured, pruning can occur when new resources are created. If you click 'refresh,' it triggers a synchronization, but 'pruning' specifically refers to the process of deleting resources in the target environment that are no longer specified in the Git repository. I also use pruning when I make changes to my Git repository and want the synchronization to occur faster. While you can achieve synchronization with a 'refresh,' using 'prune' is a more efficient way to achieve faster synchronization.
